core-tests/jvm/src/test/scala/zio/stm/StmIssue1587Spec.scala

```diff
@@ -0,0 +1,65 @@
+package zio.stm
+
+import zio.{ TestRuntime, ZIO }
+import scala.util.control.Breaks._
+
+class StmIssue1587Spec(implicit ee: org.specs2.concurrent.ExecutionEnv) extends TestRuntime {
+
+  def is = "STM made of read only operations".title ^ s2"""
+     doen't commit when journal is invalid $getOnlyStmCheck
+  """
+
+  def getOnlyStmCheck = {
+    val iterations = 1000
+    val parallel   = 50
+    val total      = iterations * parallel
+    var badResult  = Option.empty[String]
+    breakable {
+      for (i <- 1 to iterations) {
+        unsafeRun(
+          ZIO.sequencePar(Seq.fill(parallel)(test))
+        ).zipWithIndex.foreach {
+          case (result, j) =>
+            if (result != 0 && result != 5) {
+              val iteration = (i - 1) * parallel + j + 1
+              badResult = Some(s"Result[$iteration/$total] $result")
+              break()
+            }
+        }
+      }
+    }
+    badResult aka "badResult" must beNone
+  }
+
+  val test = {
+    for {
+      r0 <- TRef.makeCommit(0)
+      r1 <- TRef.makeCommit(0)
+      r2 <- TRef.makeCommit(0)
+      r3 <- TRef.makeCommit(0)
+      r4 <- TRef.makeCommit(0)
+      sum1Fiber <- r0.get.flatMap { v0 =>
+                    r1.get.flatMap { v1 =>
+                      r2.get.flatMap { v2 =>
+                        r3.get.flatMap { v3 =>
+                          r4.get.map { _ + v0 + v1 + v2 + v3 }
+                        }
+                      }
+                    }
+                  }.commit.fork
+      _ <- r0
+            .update(_ + 1)
+            .flatMap { _ =>
+              r1.update(_ + 1).flatMap { _ =>
+                r2.update(_ + 1).flatMap { _ =>
+                  r3.update(_ + 1).flatMap { _ =>
+                    r4.update(_ + 1)
+                  }
+                }
+              }
+            }
+            .commit
+      sum1 <- sum1Fiber.join
+    } yield sum1
+  }
+}
```

Comment:
I admit `val test` ~isn't pretty~ is ugly but I wanted to use as small number of functions as possible.